### PR TITLE
Fix map_keys and map_values presto function

### PR DESCRIPTION
**Pull request by bikramSingh91**
Currently both these functions generate a result array of size equal
to the total size of the selectivity vector passed to it. However if
the last valid row in the selectivity vector is smaller than its
total size (that is rows.end() < rows.size()) then this can be
larger than the size of the underlying buffers and can result in an
inconsistent state where the size of the offset or sizes buffer is
insufficient to hold all indices. This patch fixes the
aforementioned bug.

Test Plan
Added a unit test that would fail without this fix
(but needs #3078 for consistency check)

### CODE REVIEW

**Comment by bikramSingh91** [comment]
fixes part of #3043

File: velox/expression/tests/ExprTest.cpp
Comment (mbasmanova) [inline velox/expression/tests/ExprTest.cpp:2992]: Tests using Fuzzer are a bit opaque. I wonder if this issue can be reproduced using hard-coded data?

```
TEST_F(ExprTest, repro) {
  auto offsets = makeIndices(50, [](auto row) { return 0; });
  auto sizes = makeIndices(50, [](auto row) { return 0; });
  VectorPtr mapVector = std::make_shared<MapVector>(
      pool_.get(),
      MAP(BIGINT(), BIGINT()),
      makeNulls(100, [](auto row) { return row > 50; }),
      100,
      offsets,
      sizes,
      makeFlatVector<int64_t>({1, 2, 3}),
      makeFlatVector<int64_t>({10, 20, 30}));

  evaluate("map_keys(c0)", makeRowVector({mapVector}));
  evaluate("map_values(c0)", makeRowVector({mapVector}));
}
```

**Review by mbasmanova — APPROVED** [review]
@bikramSingh91 Looks great. Thank you for the fix. I have a comment re: test.

File: velox/expression/tests/ExprTest.cpp
Comment (bikramSingh91) [reply velox/expression/tests/ExprTest.cpp:2992]: Unfortunately the test case that you shared above will hit the consistency check in #3078 while creating the mapVector (offset.size() < 100).
Therefore to hit the check inside map_keys/values instead we need the generate a map smaller than rows.size() during evaluation, which I achieved by wrapping a map vector in lazy then materializing (Size-1) part of it.

However, while writing this reply I just realized I can achieve the same by creating a map of size 100 and a SelectivityVector of size 101 (marking the 100th idx as invalid) then executing the eval. thereby avoiding all the fuzzer + lazy hassle. So, I'll post the simplified version soon

**Comment by bikramSingh91** [comment]
will wait for #3078 to be merged before merging this to make sure I get the right signals from pre-commits

**Pull request closed without merging**
